app/controllers/user.js

```diff
@@ -3,28 +3,50 @@ const { getOwner } = Ember;
 
 export default Ember.Controller.extend({
   user: Ember.computed.alias('model'),
-  selectedId: null,
+  selectedRoleIds: [],
 
   permissions: Ember.computed(function(){
     return this.store.peekAll("permission").rejectBy("name", "System").sortBy('name');
   }),
 
+  roles: Ember.computed(function(){
+    return this.store.peekAll("role");
+  }),
+
   actions: {
+    setSelecteIds(id, isSelected) {
+      if(isSelected){
+        this.get('selectedRoleIds').pushObject(id);
+      } else {
+        this.get('selectedRoleIds').removeObject(id);
+      }
+    },
+
     saveUser(){
+      var store = this.store;
       var user = this.get("model");
```

Comment:
Instead of using **this.get("model");** can we use **this.get('user')** as It's declared at the beginning => **user: Ember.computed.alias('model'),**.
Or may be we can remove **user: Ember.computed.alias('model'),**?